packages/text-bitmap/src/BitmapFont.ts

```diff
@@ -226,4 +236,298 @@ export class BitmapFont
     {
         return available;
     }
+
+    /**
+     * Generates a bitmap-font for the given style and character set. This does not support
+     * kernings yet.
+     *
+     * @param {PIXI.IBitmapFontFactoryOptions} options - includes all `ITextStyle` properties.
+     * @param {string|string[]} options.chars - characters included in the font set. You can also
+     *      use ranges. For example, `[['a', 'z'], ['A', 'Z'], "!@#$%^&*()~`{}[]"]. Don't forget to
+     *      include spaces ' ' in your character set!
+     * @param {number} [options.resolution=1] - optional resolution to render the glyphs at
+     * @return {PIXI.BitmapFont} font generated by style options.
+     * @static
+     * @example
+     * const font = PIXI.BitmapFont.from({
+     *     chars: [['a', 'z'], ['A', 'Z'], "!#$%^&*()~`{}[]"]
```

Comment:
It'd be nice to provide either some defaults, or some guidance for defaults. Like
chars: 'ASCII' to auto add all ascii chars, or docs in here to say 'here is the regex to add all ascii characters
chars: 'numeric' to add all numbers plus known separators, or docs in here to say 'here is the regex to add all numbers'
Docs probably the easier route, but I think getting in some common scenarios now would save lots of questions in the future

Reply:
BitmapFont.ALPHA
BitmapFont.NUMERIC
BitmapFont.ALPHA_NUMERIC

Convenience things like that?

Reply:
Yeah. They refer to regexes themselves in the background. We could have a collection of those, so I could go

options.chars = [ BitmapFont.ALPHA_UPPERCASE, BitmapFont.NUMERIC, BitmapFont.SEPARATORS ]

If I wanted chars just for a high score board for example

Reply:
I’m also okay with a follow up PR to add some syntax sugar on top of setting character ranges. I don’t think it changes the base functionality here.

Reply:
@themoonrat I added `ALPHA`, `NUMERIC`, `ALPHANUMERIC`, and `ASCII`. @bigtimebuddy Do you think we should add more (like lower/uppercase, separators, ...)?

Reply:
I think that’s okay for now. We can add more easily if need be. Let’s see what kinda feedback we get once we release this.